### DISCUSSION TITLE
Nats Operator & Sensor

### DISCUSSION THREAD

**Discussion by khoramism** [apache/airflow#47391] [Ideas]
### Description

I think a lot of the teams already use nats as their message broker or streaming services and it would be a great Idea to have this as a part of the operators in airflow as well.

### Use case/motivation

What I've been trying to achieve is the same process with the [kafka](https://airflow.apache.org/docs/apache-airflow-providers-apache-kafka/stable/index.html), a Producer and a Consumer working together to implement or do something. 
For example in my case I tried to use airflow as an Alerting system with Kafka, I'd do some stuff using different operators and at the end of the day, I'd just push all the anomalies or errors to a topic in Kafka, then I'd use Kafka and telegram or slack operators to consume from the topic and send the messages needed to a specific channel or ...

And It would just be a great idea to have the same process and support out of the box for [Nats](https://nats.io/) as well as [Kafka](https://kafka.apache.org/) 

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by Taragolis** [comment]
Please note that all new Community supported providers should comes thought [Accepting new community providers](https://github.com/apache/airflow/blob/main/PROVIDERS.rst#id5) process.

**Comment by static-moonlight** [comment]
Yeah. We really need that! It makes me sad to see that nothing happened here in over 2 years.

There are a bunch of [systems](https://airflow.apache.org/docs/apache-airflow-providers/core-extensions/message-queues.html), which are supported already. I find it interesting though that even typical messaging protocols like AMQP or MQTT are not supported yet. However, NATS definitely deserves to be put on that list.

@Taragolis

Honestly, it doesn't make any sense that even proposing the idea of NATS support in Airflow has to immediately start with the acceptance process, since the first [rule](https://github.com/apache/airflow/blob/main/providers/ACCEPTING_PROVIDERS.rst#prerequisites-for-proposing-a-new-provider) to start that process is to have a working code base already. Basically saying: "You can't talk about NATS support here unless you bring the code"?
